### PR TITLE
Delete all unused private methods

### PR DESCRIPTION
**Pull request by original-brownbear**
Pretty straight forward automated dead-code cleanup I think. Just delete all private methods or methods in private classes that aren't used.

### CODE REVIEW

**Comment by ldematte** [comment]
Hi! QQ from a learner: I started going through files for a quick check of the changes, but questions started to raise in my mind. Now, since these are private methods (or public methods in private classes), if  the code compiles and tests run successfully it _should_ be OK, right? Or there could be subtle problems, e.g. usage by reflection (including serialization/deserialization hooks, for example)? I'm new to the codebase, so I'm not sure if these things could be happening/ are allowed or not.

**Comment by original-brownbear** [comment]
Thanks for taking a look @ldematte !

> if the code compiles and tests run successfully it should be OK, right? 

That was my assumption yes.

> ? Or there could be subtle problems, e.g. usage by reflection (including serialization/deserialization hooks, for example)

The only place that I know of that uses a bit of dynamic logic is our own scripting language painless. There we use method handles derived from private methods. I did this refactoring using Jetbrains Idea though and it is clever enough to figure out references via method handle lookups. Also, even if it wasn't, static initialiser method handle lookups should fail at class load time reliably.

-> I don't know of any reflection or so in the codebase that could hurt us here at least :)

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java [hunk 1]
```diff
@@ -1096,51 +1093,6 @@ protected LogicalPlan rule(OrderBy ob) {
         }
     }
 
-    private static class ImplicitCasting extends AnalyzerRule<LogicalPlan> {
```
Comment (ldematte) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java:1099]: If you want to remove this class, you should probably remove its instantiation at line 100 (`// new ImplicitCasting()`)

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java [hunk 2]
```diff
@@ -469,6 +469,4 @@ IngestStats.Stats build() {
         }
     }
 
-    private record ModelAndDeployment(String modelId, String deploymentId) {}
```
Comment (ldematte) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetTrainedModelsStatsAction.java:472]: This is a record, not really a private method :) but it looks unused anyway

**Review by ldematte — APPROVED** [review]
I checked the changes in this way:
- For test files (test case, IT, unit tests) I have just checked removal looked OK, quickly (40 files)
- For the other cases (19) I decided case by case what to do:
    - private inner class completely unused, probably safe
    - private or public methods of private static class, probably safe
    - Etc.
In most cases I also did a quick search on the Elasticsearch repo (but only on that) to see if there were text usages (e.g. painless whitelisting) of the method/class.

LGTM but it is probably wise to get a second opinion

**Comment by original-brownbear** [comment]
Thanks Lorenzo, nobody ever commented here. I backed out the few controversial points you found -> merging :)

**Comment by ldematte** [comment]
Thanks for the cleanup!

**Pull request merged**